backend/src/middlewares/errorHandler.ts

```diff
@@ -3,33 +3,81 @@ import { Request, Response, NextFunction } from "express";
 
 import CustomError from "./customError";
 interface ErrorObject {
+  message: string;
+  kind: string;
+  keyValue: any;
+  code: number;
   name: string;
   details?: any;
   statusCode: number;
   errorCode: number | string;
 }
 
+const handleValidationError = (error: ErrorObject, res: Response) => {
+  return res.status(400).json({
+    type: "ValidationError",
+    errorMessage: error.details[0].message,
+  });
+};
+
+const handleNotFoundError = (error: ErrorObject, res: Response) => {
+  return res.status(404).json({
+    type: "NotFoundError",
+    errorMessage: error.details,
+  });
+};
+
+const handleDuplicateError = (error: ErrorObject, res: Response) => {
+  return res.status(400).json({
+    type: "DuplicateError",
+    errorMessage: `duplicate ${Object.keys(error.keyValue)}`,
+  });
+};
+
+const handleObjectIdError = (error: ErrorObject, res: Response) => {
+  return res.status(400).json({
+    type: "InvalidDocumentId",
+    errorMessage: error.message,
+  });
+};
+
+const handleCustomError = (error: ErrorObject, res: Response) => {
+  return res.status(error.statusCode).json({
+    type: error.errorCode,
+    errorMessage: error.message,
+  });
+};
+
+const handleServerError = (res: Response) => {
+  res.status(500).json({
+    type: "ServerError",
+    errorMessage: "Something Went Wrong. Our team is working on it",
+  });
+};
+
 export const errorHandler = (
   error: ErrorObject,
   req: Request,
   res: Response,
   next: NextFunction,
 ) => {
-  if (error.name === "ValidationError") {
-    return res.status(400).send({
-      type: "ValidationError",
-      details: error.details,
-    });
+  switch (true) {
```

Comment:
Hi @NNesta this is a bit different of what I've requested, maybe we should have a short call to clarify this.

Reply:
Sure we should have it. You can tell us a time you are free.

Reply:
I was uber busy and could not have that call, but I manage now to at least approve your PR :) .. if you still want we can have that discussion maybe sometime next week?
But I am not sure if slack is still working or not